overlord/snapshotstate/backend/backend.go

```diff
@@ -336,7 +336,8 @@ func Save(ctx context.Context, id uint64, si *snap.Info, cfg map[string]interfac
 	w := zip.NewWriter(aw)
 	defer w.Close() // note this does not close the file descriptor (that's done by hand on the atomic writer, above)
 	savingUserData := false
-	if err := addDirToZip(ctx, snapshot, w, "root", archiveName, si.DataDir(), savingUserData, snapshotOptions.ExcludePaths); err != nil {
+	baseDataDir := snap.BaseDataDir(si.InstanceName())
```

Comment:
AFAIU, previously we would add DataDir for a specific revision only, while now we're adding entire BaseDataDir of the snap. Is this right/intended?

Reply:
I think it's a bug. Thank you for spotting it, I've fixed it now. It's odd that the tests didn't catch that, I'll have to look at that

Reply:
Sorry, my last comment was completely idiotic. I did this a few months ago so I honestly sort of forgot what motivated it to begin with. Yes, the change is intended and the tests are correct. 
Essentially, what the previous code was doing was passing the revisioned dir (both for root and the user) to `addDirToZip`. That function would then check if the revisioned and common dirs existed through a longer and more contrived version of:
```
... = osutil.DirExists(dir)
...
parent, revdir := filepath.Split(dir)
common := filepath.Join(parent, "common")
... = osutil.DirExists(common)
...
```
So the new version just passes the base (parent) directory to a function that encapsulates the handling of "snap" dirs and generates the right paths for the snapshot. Then those are passed to `addToZip` which only contains logic for actually creating the snapshot (i.e., it doesn't care about internal snap directory details). I'll revert my last commit.